examples/cloud-operations/network-dashboard/cloud-function/main.py

```diff
@@ -59,39 +63,192 @@ def main(event, context):
       Returns:
         'Function executed successfully'
   '''
-
   metrics_dict = create_metrics()
 
+  # Asset inventory queries
+  gce_instance_dict = get_gce_instance_dict()
+  l4_forwarding_rules_dict = get_l4_forwarding_rules_dict()
+  l7_forwarding_rules_dict = get_l7_forwarding_rules_dict()
+  subnet_range_dict = get_subnet_ranges_dict()
+
   # Per Network metrics
-  get_gce_instances_data(metrics_dict)
-  get_l4_forwarding_rules_data(metrics_dict)
+  get_gce_instances_data(metrics_dict, gce_instance_dict)
+  get_l4_forwarding_rules_data(metrics_dict, l4_forwarding_rules_dict)
   get_vpc_peering_data(metrics_dict)
 
   get_pgg_data(
       metrics_dict["metrics_per_peering_group"]["instance_per_peering_group"],
-      GCE_INSTANCES_USAGE_METRIC, GCE_INSTANCES_LIMIT_METRIC,
-      LIMIT_INSTANCES_PPG)
+      gce_instance_dict, GCE_INSTANCES_LIMIT_METRIC, LIMIT_INSTANCES_PPG)
 
   get_pgg_data(
       metrics_dict["metrics_per_peering_group"]
-      ["l4_forwarding_rules_per_peering_group"],
-      L4_FORWARDING_RULES_USAGE_METRIC, L4_FORWARDING_RULES_LIMIT_METRIC,
-      LIMIT_L4_PPG)
+      ["l4_forwarding_rules_per_peering_group"], l4_forwarding_rules_dict,
+      L4_FORWARDING_RULES_LIMIT_METRIC, LIMIT_L4_PPG)
 
   get_pgg_data(
       metrics_dict["metrics_per_peering_group"]
-      ["l7_forwarding_rules_per_peering_group"],
-      L7_FORWARDING_RULES_USAGE_METRIC, L7_FORWARDING_RULES_LIMIT_METRIC,
-      LIMIT_L7_PPG)
+      ["l7_forwarding_rules_per_peering_group"], l7_forwarding_rules_dict,
+      L7_FORWARDING_RULES_LIMIT_METRIC, LIMIT_L7_PPG)
 
   get_pgg_data(
       metrics_dict["metrics_per_peering_group"]
-      ["subnet_ranges_per_peering_group"], SUBNET_RANGES_USAGE_METRIC,
+      ["subnet_ranges_per_peering_group"], subnet_range_dict,
       SUBNET_RANGES_LIMIT_METRIC, LIMIT_SUBNETS)
 
   return 'Function executed successfully'
 
 
+def get_l4_forwarding_rules_dict():
```

Comment:
Once this PR is in, we might want to refactor the CAI `get` methods so they all use a shared `cloud_asset_search` function as there's quite a bit of duplicate code between them.

Reply:
Good point, I think this can be addressed after the PR is merged.